gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -1871,11 +1833,17 @@ def test_{{ method_name }}_rest_no_http_options():
         {% endfor %}{# for auto_populated_field in method_settings.auto_populated_fields #}
         {% endif %}{# if method_settings is not none #}
         {% endwith %}{# method_settings #}
-        {% if request %}
-        assert args[0] == {{ request }}
+        {% if request_dict %}
+        request_msg = {{ method.input.ident }}(**{{ request_dict }})
         {% else %}
-        assert args[0] == {{ method.input.ident }}()
-        {% endif %}{# request #}
+        request_msg = {{ method.input.ident }}()
+        {% endif %}{# request_dict #}      
+        assert args[0] == request_msg
+
+        {% if routing_param %}
+        expected_headers = {{ method.routing_rule.resolve(method.routing_rule, routing_param.sample_request) }}
+        assert gapic_v1.routing_header.to_grpc_metadata(expected_headers) in kw['metadata']
```

Comment:
Is this code path tested? I don't find it in any of the goldens.

Reply:
Yes, you can add an `assert False` when running the showcase unit tests and tests will fail. I attempted to add golden files for showcase in https://github.com/googleapis/gapic-generator-python/pull/1991 but ran into the issue in https://github.com/googleapis/gapic-generator-python/issues/1993